src/Dialect/ONNX/ONNXOps.cpp

```diff
@@ -3267,10 +3267,18 @@ LogicalResult ONNXQuantizeLinearOp::inferShapes(
   auto yTy = y().getType().cast<ShapedType>();
 
   if (!yTy.hasStaticShape()) {
-    // TODO: Unfortunately, we can't tell if this should be signed or
-    // unsigned here...
-    IntegerType i8Type = IntegerType::get(getContext(), 8);
-    RankedTensorType outType = RankedTensorType::get(inTy.getShape(), i8Type);
+    auto zero = y_zero_point();
+    RankedTensorType outType;
+    // If zero point type isn't provided, output type defaults to ui8.
+    if (isFromNone(zero)) {
+      auto builder = mlir::Builder(this->getContext());
+      IntegerType ui8Type = builder.getIntegerType(8, false);
```

Comment:
nit: I think the most readable and unambiguous way to express this (and gets rid of the builder) is:
```
IntegerType::get(getContext(), 8, IntegerType::Unsigned)
```

Reply:
Thanks! I was struggling to find `IntegerType` in the mlir docs. I saw the function on the builder.

But your suggestion is definitely better.